PseudoGantry/lib/PseudoGantry/PseudoAxis.cpp

```diff
@@ -0,0 +1,121 @@
+#include "PseudoAxis.h"
+
+void set_up_encoder(PseudoEncoder *encoder, void (*isr_motor_pulse)(void))
+{
+    pinMode(encoder->motor_pulse_pin, INPUT_PULLUP);
+    pinMode(encoder->channel_a_out, OUTPUT);
+    digitalWrite(encoder->channel_a_out, LOW);
+    attachInterrupt(digitalPinToInterrupt(encoder->motor_pulse_pin), isr_motor_pulse, CHANGE);
+}
+
+bool toggle_encoder_output(PseudoAxis *pseudo_axis)
+{
+    pseudo_axis->skip_counter++;
+    // skip steps in the beginning
+    if (pseudo_axis->skip_counter <= pseudo_axis->changes_to_skip) return false;
+    // else start toggling until the next set of STEPS
+    else if ((pseudo_axis->skip_counter <= (2 * pseudo_axis->steps_for_ratio)) || (pseudo_axis->changes_to_skip == 0))
+    {   
+        bool motor_pin_status = digitalRead(pseudo_axis->encoder.motor_pulse_pin);
```

Comment:
`digitalRead` returns an `int` so I would declare this as an `int` or explicitly cast it